src/main/core/support/configuration.rs

```diff
@@ -16,40 +524,230 @@ pub enum InterposeMethod {
 }
 
 impl std::str::FromStr for InterposeMethod {
-    type Err = String;
+    type Err = serde_yaml::Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match s.to_lowercase().as_str() {
-            "ptrace" => Ok(Self::Ptrace),
-            "preload" => Ok(Self::Preload),
-            "hybrid" => Ok(Self::Hybrid),
-            _ => Err(format!("'{}' is not a valid interpose method", s)),
+        serde_yaml::from_str(s)
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "lowercase")]
+pub enum SchedulerPolicy {
+    Thread,
+    Host,
+    Steal,
+    ThreadXThread,
+    ThreadXHost,
+}
+
+impl std::str::FromStr for SchedulerPolicy {
+    type Err = serde_yaml::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        serde_yaml::from_str(s)
+    }
+}
+
+impl std::fmt::Display for SchedulerPolicy {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            Self::Thread => write!(f, "thread"),
```

Comment:
Optional - `#[derive(Display)]`? https://docs.rs/enum-display-derive/0.1.1/enum_display_derive/

Reply:
I removed the `impl std::fmt::Display for SchedulerPolicy` here and instead added a function that converts directly to the type `c::SchedulerPolicyType`.